tests/test_evaluate.py

```diff
@@ -171,11 +171,37 @@ def test_run_evaluation():
 
 
 class MockResponse:
-    def __init__(self, json_data, status_code=200, reason="Fail Test"):
+    def __init__(self, json_data, status_code=200, reason="Fail Test", url="http://mock-url.com"):
         self.json_data = json_data
         self.status_code = status_code
         self.reason = reason
         self.elapsed = timedelta(seconds=1)
+        self.url = url
+
+    def raise_for_status(self):
+        """Raises :class:`HTTPError`, if one occurred."""
+
+        http_error_msg = ""
+        if isinstance(self.reason, bytes):
+            # We attempt to decode utf-8 first because some servers
+            # choose to localize their reason strings. If the string
+            # isn't utf-8, we fall back to iso-8859-1 for all other
+            # encodings. (See PR #3538)
+            try:
+                reason = self.reason.decode("utf-8")
+            except UnicodeDecodeError:
+                reason = self.reason.decode("iso-8859-1")
+        else:
+            reason = self.reason
+
+        if 400 <= self.status_code < 500:
+            http_error_msg = f"{self.status_code} Client Error: {reason} for url: {self.url}"
+
+        elif 500 <= self.status_code < 600:
+            http_error_msg = f"{self.status_code} Server Error: {reason} for url: {self.url}"
+
+        if http_error_msg:
+            raise requests.HTTPError(http_error_msg, response=self)
```

Comment:
@GrayNekoBean why do we copy-paste the [actual implementation of the method](https://github.com/psf/requests/blob/0e322af87745eff34caffe4df68456ebc20d9068/src/requests/models.py#L997-L1024) in a mocked response? If we wanted the actual thing to run, we wouldn't really mock it, would we?

On a side note, when copy-pasting stuff like that, I expect to find an explicit reference in the commit message and possibly in a code comment, pointing to the source.

Reply:
When I committed it I realized you might have some comment against these lines, but I couldn't thought of any better ideas. My thought is to mock how it actually behaves as a Response object even it's a mocked one. I'll add a comment here that states the original source of this snippet in next commit.